Add rolling upgrade interstitials to UA

cjcenizal opened the pull request:
Fixes https://github.com/elastic/kibana/issues/111808

Changes:
* Added isClusterUpgrading and isClusterUpgraded flags to context and appropriate UI logic to handle these states.

Wrapped API routes:
- [x] `cloud_backup_status.ts` Retrieve backup status on Cloud.
- [x] `deprecation_logging.ts` Set and retrieve deprecation logging status and retrieve log count.
- [x] `es_deprecations.ts` Retrieve ES deprecations and their reindexing status.
- [x] `ml_snapshots.ts` Upgrade and delete ML snapshots, and retrieve their upgrading status.
- [x] `status.ts` Provide status to Cloud to determine whether the cluster is ready to be upgraded.
- [x] `update_index_settings.ts` Fix a deprecation by deleting an index's settings.
- [x] `reindex.indices.ts` Reindex individual indices.
- [x] `batch_reindex.indices.ts` Reindex batches of indices.

We can safely ignore these API routes:
- `app.ts` This retrieves privileges to prevent unauthorized users from accessing UA. As an aside, this file could probably be named more clearly to reflect this role.

## How to test

The interstitial is triggered whenever the above APIs respond with a `426` error. You can force this error by updating [`es_version_precheck.ts`](https://github.com/elastic/kibana/blob/7.x/x-pack/plugins/upgrade_assistant/server/lib/es_version_precheck.ts#L104) on line 104:

```ts
export const versionCheckHandlerWrapper =
  <P, Q, B>(handler: RequestHandler<P, Q, B>, isFailure) =>
  async (
    ctx: RequestHandlerContext,
    request: KibanaRequest<P, Q, B>,
    response: KibanaResponseFactory
  ) => {
    if (isFailure) {
      return response.customError({
        // 426 means "Upgrade Required" and is used when semver compatibility is not met.
        statusCode: 426,
        body: {
          message: 'There are some nodes running a different version of Elasticsearch',
          attributes: {
            allNodesUpgraded: false,
          },
        },
      });
    }
    const errorResponse = await esVersionCheck(ctx, response);
    if (errorResponse) {
      return errorResponse;
    }
    return handler(ctx, request, response);
  };
```

Then just pass `true` as the second argument to any API route handler that wraps its logic in `versionCheckHandlerWrapper()`. You can switch between the different interstitial states by changing `allNodesUpgraded` to `true`.

## Screenshots

When the cluster is upgrading:

![image](https://user-images.githubusercontent.com/1238659/134434146-b3e591a1-8929-492f-80bf-5263e4aeb9a4.png)

When the upgrade is complete:

![image](https://user-images.githubusercontent.com/1238659/134434149-d4e00f14-80d4-41fa-a922-f848d64ea647.png)

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155162/)
* Commit: c4964e53c1be908cbb9f0862a0ee1903f1f474cd
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types and Public API Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155162/execution/node/428/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155162/","number":"155162","commit":"c4964e53c1be908cbb9f0862a0ee1903f1f474cd"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155496/)
* Commit: 708077d90bb2c0fe3aeaed8ced9359c7e18cf76c
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Quick Commit Checks](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155496/execution/node/421/log)

### History
* :broken_heart: [Build #155162](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155162/) failed c4964e53c1be908cbb9f0862a0ee1903f1f474cd

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155162/","number":"155162","commit":"c4964e53c1be908cbb9f0862a0ee1903f1f474cd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155496/","number":"155496","commit":"708077d90bb2c0fe3aeaed8ced9359c7e18cf76c"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156897/)
* Commit: 70975ab79435d526307096e1843959f58d6a16f3
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types and Public API Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156897/execution/node/428/log)

### History
* :broken_heart: [Build #155496](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155496/) failed 708077d90bb2c0fe3aeaed8ced9359c7e18cf76c
* :broken_heart: [Build #155162](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155162/) failed c4964e53c1be908cbb9f0862a0ee1903f1f474cd

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155162/","number":"155162","commit":"c4964e53c1be908cbb9f0862a0ee1903f1f474cd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155496/","number":"155496","commit":"708077d90bb2c0fe3aeaed8ced9359c7e18cf76c"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156897/","number":"156897","commit":"70975ab79435d526307096e1843959f58d6a16f3"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157241/)
* Commit: eb5c8da9a74953d7152cf63536d0d447faadc97e
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Quick Commit Checks](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157241/execution/node/401/log)

### History
* :broken_heart: [Build #156897](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156897/) failed 70975ab79435d526307096e1843959f58d6a16f3
* :broken_heart: [Build #155496](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155496/) failed 708077d90bb2c0fe3aeaed8ced9359c7e18cf76c
* :broken_heart: [Build #155162](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155162/) failed c4964e53c1be908cbb9f0862a0ee1903f1f474cd

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155162/","number":"155162","commit":"c4964e53c1be908cbb9f0862a0ee1903f1f474cd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155496/","number":"155496","commit":"708077d90bb2c0fe3aeaed8ced9359c7e18cf76c"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156897/","number":"156897","commit":"70975ab79435d526307096e1843959f58d6a16f3"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157241/","number":"157241","commit":"eb5c8da9a74953d7152cf63536d0d447faadc97e"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157257/)
* Commit: cf59731eeaef2ed2fc7df20fe6970f77dcf56812
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types and Public API Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157257/execution/node/428/log)

### History
* :broken_heart: [Build #157241](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157241/) failed eb5c8da9a74953d7152cf63536d0d447faadc97e
* :broken_heart: [Build #156897](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156897/) failed 70975ab79435d526307096e1843959f58d6a16f3
* :broken_heart: [Build #155496](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155496/) failed 708077d90bb2c0fe3aeaed8ced9359c7e18cf76c
* :broken_heart: [Build #155162](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155162/) failed c4964e53c1be908cbb9f0862a0ee1903f1f474cd

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155162/","number":"155162","commit":"c4964e53c1be908cbb9f0862a0ee1903f1f474cd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155496/","number":"155496","commit":"708077d90bb2c0fe3aeaed8ced9359c7e18cf76c"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156897/","number":"156897","commit":"70975ab79435d526307096e1843959f58d6a16f3"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157241/","number":"157241","commit":"eb5c8da9a74953d7152cf63536d0d447faadc97e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157257/","number":"157257","commit":"cf59731eeaef2ed2fc7df20fe6970f77dcf56812"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157259/)
* Commit: b814900e48a4f3c74d29b374e7d24aa2b74a68da
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types and Public API Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157259/execution/node/422/log)

### History
* :broken_heart: [Build #157257](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157257/) failed cf59731eeaef2ed2fc7df20fe6970f77dcf56812
* :broken_heart: [Build #157241](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157241/) failed eb5c8da9a74953d7152cf63536d0d447faadc97e
* :broken_heart: [Build #156897](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156897/) failed 70975ab79435d526307096e1843959f58d6a16f3
* :broken_heart: [Build #155496](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155496/) failed 708077d90bb2c0fe3aeaed8ced9359c7e18cf76c
* :broken_heart: [Build #155162](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155162/) failed c4964e53c1be908cbb9f0862a0ee1903f1f474cd

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155162/","number":"155162","commit":"c4964e53c1be908cbb9f0862a0ee1903f1f474cd"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155496/","number":"155496","commit":"708077d90bb2c0fe3aeaed8ced9359c7e18cf76c"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156897/","number":"156897","commit":"70975ab79435d526307096e1843959f58d6a16f3"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157241/","number":"157241","commit":"eb5c8da9a74953d7152cf63536d0d447faadc97e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157257/","number":"157257","commit":"cf59731eeaef2ed2fc7df20fe6970f77dcf56812"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157259/","number":"157259","commit":"b814900e48a4f3c74d29b374e7d24aa2b74a68da"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157271/)
* Commit: 890e9addc352e5d136a8e3bad2da2b13f24dcc66
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types and Public API Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157271/execution/node/428/log)

### History
* :broken_heart: [Build #157241](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157241/) failed eb5c8da9a74953d7152cf63536d0d447faadc97e
* :broken_heart: [Build #157259](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157259/) failed b814900e48a4f3c74d29b374e7d24aa2b74a68da
* :broken_heart: [Build #157257](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157257/) failed cf59731eeaef2ed2fc7df20fe6970f77dcf56812
* :broken_heart: [Build #156897](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156897/) failed 70975ab79435d526307096e1843959f58d6a16f3
* :broken_heart: [Build #155496](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155496/) failed 708077d90bb2c0fe3aeaed8ced9359c7e18cf76c

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/155496/","number":"155496","commit":"708077d90bb2c0fe3aeaed8ced9359c7e18cf76c"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156897/","number":"156897","commit":"70975ab79435d526307096e1843959f58d6a16f3"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157257/","number":"157257","commit":"cf59731eeaef2ed2fc7df20fe6970f77dcf56812"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157259/","number":"157259","commit":"b814900e48a4f3c74d29b374e7d24aa2b74a68da"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157241/","number":"157241","commit":"eb5c8da9a74953d7152cf63536d0d447faadc97e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157271/","number":"157271","commit":"890e9addc352e5d136a8e3bad2da2b13f24dcc66"}]}
PIPELINE-->

cjcenizal commented on x-pack/plugins/upgrade_assistant/public/application/components/overview/fix_logs_step/fix_logs_step.tsx:
```diff
@@ -134,10 +145,18 @@ const FixLogsStep: FunctionComponent<Props> = ({
       </EuiText>
       <EuiSpacer size="m" />
       <EuiPanel>
-        <DeprecationLoggingToggle {...state} />
```
By passing through the entire object returned by `useDeprecationLogging` we create a tight coupling between that hook's return value and `DeprecationLoggingToggle` props, which creates brittleness if we make changes to either one.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157276/)
* Commit: 05eb0856313084789b8d7eccd7c4a8c57a2c739e
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types and Public API Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157276/execution/node/428/log)

### History
* :broken_heart: [Build #157271](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157271/) failed 890e9addc352e5d136a8e3bad2da2b13f24dcc66
* :broken_heart: [Build #157241](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157241/) failed eb5c8da9a74953d7152cf63536d0d447faadc97e
* :broken_heart: [Build #157259](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157259/) failed b814900e48a4f3c74d29b374e7d24aa2b74a68da
* :broken_heart: [Build #157257](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157257/) failed cf59731eeaef2ed2fc7df20fe6970f77dcf56812
* :broken_heart: [Build #156897](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156897/) failed 70975ab79435d526307096e1843959f58d6a16f3

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/156897/","number":"156897","commit":"70975ab79435d526307096e1843959f58d6a16f3"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157257/","number":"157257","commit":"cf59731eeaef2ed2fc7df20fe6970f77dcf56812"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157259/","number":"157259","commit":"b814900e48a4f3c74d29b374e7d24aa2b74a68da"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157241/","number":"157241","commit":"eb5c8da9a74953d7152cf63536d0d447faadc97e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157271/","number":"157271","commit":"890e9addc352e5d136a8e3bad2da2b13f24dcc66"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157276/","number":"157276","commit":"05eb0856313084789b8d7eccd7c4a8c57a2c739e"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157586/)
* Commit: 94a052da026ab6c65474cb1da72399230b1728ea
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types and Public API Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157586/execution/node/409/log)

### History
* :broken_heart: [Build #157271](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157271/) failed 890e9addc352e5d136a8e3bad2da2b13f24dcc66
* :broken_heart: [Build #157276](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157276/) failed 05eb0856313084789b8d7eccd7c4a8c57a2c739e
* :broken_heart: [Build #157259](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157259/) failed b814900e48a4f3c74d29b374e7d24aa2b74a68da
* :broken_heart: [Build #157241](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157241/) failed eb5c8da9a74953d7152cf63536d0d447faadc97e
* :broken_heart: [Build #157257](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157257/) failed cf59731eeaef2ed2fc7df20fe6970f77dcf56812

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157257/","number":"157257","commit":"cf59731eeaef2ed2fc7df20fe6970f77dcf56812"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157241/","number":"157241","commit":"eb5c8da9a74953d7152cf63536d0d447faadc97e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157259/","number":"157259","commit":"b814900e48a4f3c74d29b374e7d24aa2b74a68da"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157276/","number":"157276","commit":"05eb0856313084789b8d7eccd7c4a8c57a2c739e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157271/","number":"157271","commit":"890e9addc352e5d136a8e3bad2da2b13f24dcc66"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157586/","number":"157586","commit":"94a052da026ab6c65474cb1da72399230b1728ea"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157731/)
* Commit: 01e3ede115bead1deb7f47f30ba45bbffed40456
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types and Public API Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157731/execution/node/433/log)

### History
* :broken_heart: [Build #157586](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157586/) failed 94a052da026ab6c65474cb1da72399230b1728ea
* :broken_heart: [Build #157271](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157271/) failed 890e9addc352e5d136a8e3bad2da2b13f24dcc66
* :broken_heart: [Build #157276](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157276/) failed 05eb0856313084789b8d7eccd7c4a8c57a2c739e
* :broken_heart: [Build #157259](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157259/) failed b814900e48a4f3c74d29b374e7d24aa2b74a68da
* :broken_heart: [Build #157241](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157241/) failed eb5c8da9a74953d7152cf63536d0d447faadc97e

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157241/","number":"157241","commit":"eb5c8da9a74953d7152cf63536d0d447faadc97e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157259/","number":"157259","commit":"b814900e48a4f3c74d29b374e7d24aa2b74a68da"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157276/","number":"157276","commit":"05eb0856313084789b8d7eccd7c4a8c57a2c739e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157271/","number":"157271","commit":"890e9addc352e5d136a8e3bad2da2b13f24dcc66"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157586/","number":"157586","commit":"94a052da026ab6c65474cb1da72399230b1728ea"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157731/","number":"157731","commit":"01e3ede115bead1deb7f47f30ba45bbffed40456"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157788/)
* Commit: c9da569bf509ea9cdede114600be7b525b5ee081
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types and Public API Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157788/execution/node/428/log)

### History
* :broken_heart: [Build #157731](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157731/) failed 01e3ede115bead1deb7f47f30ba45bbffed40456
* :broken_heart: [Build #157586](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157586/) failed 94a052da026ab6c65474cb1da72399230b1728ea
* :broken_heart: [Build #157271](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157271/) failed 890e9addc352e5d136a8e3bad2da2b13f24dcc66
* :broken_heart: [Build #157276](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157276/) failed 05eb0856313084789b8d7eccd7c4a8c57a2c739e
* :broken_heart: [Build #157259](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157259/) failed b814900e48a4f3c74d29b374e7d24aa2b74a68da

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157259/","number":"157259","commit":"b814900e48a4f3c74d29b374e7d24aa2b74a68da"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157276/","number":"157276","commit":"05eb0856313084789b8d7eccd7c4a8c57a2c739e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157271/","number":"157271","commit":"890e9addc352e5d136a8e3bad2da2b13f24dcc66"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157586/","number":"157586","commit":"94a052da026ab6c65474cb1da72399230b1728ea"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157731/","number":"157731","commit":"01e3ede115bead1deb7f47f30ba45bbffed40456"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/157788/","number":"157788","commit":"c9da569bf509ea9cdede114600be7b525b5ee081"}]}
PIPELINE-->

cjcenizal commented:
Thanks for the reviews @yuliacech and @sabarasaba! I've addressed your comments and asked a couple questions for clarification. Could you please take another look?

@yuliacech I named one of the components in `app.tsx` to `AppHandlingClusterUpgradeState` and added more detailed comments to explain the split. I left the other component named `App` because I think changing its name will leak an implementation detail to the tests, which consume it. Can you take a look and let me know if this is enough to clarify things?

cjcenizal commented:
@yuliacech And thanks for the tip about tweak!
